app/src/apdu_handler.c

```diff
@@ -31,12 +31,45 @@
 #include "zxmacros.h"
 
 __Z_INLINE void handleGetPubkey(volatile uint32_t *flags, volatile uint32_t *tx, uint32_t rx) {
-    extractHDPath(rx, OFFSET_DATA);
+    if (rx != 6) {
+        THROW(APDU_CODE_DATA_INVALID);
+    }
 
     uint8_t requireConfirmation = G_io_apdu_buffer[OFFSET_P1];
+    const uint8_t slotIdx = G_io_apdu_buffer[OFFSET_DATA];
+
+    char buffer[20];
+    snprintf(buffer, sizeof(buffer), "%d", slotIdx);
+    zemu_log_stack(buffer);
+
+    account_slot_t slot;
+    zxerr_t err = slot_getSlot(slotIdx,(uint8_t *) &slot, sizeof(slot));
+    snprintf(buffer, sizeof(buffer), "err: %d", err);
+    zemu_log_stack(buffer);
+
+    if (err == zxerr_no_data) {
+        zemu_log_stack("Empty slot");
+        THROW(APDU_CODE_DATA_INVALID);
+    }
+
+    if (err != zxerr_ok) {
+        THROW(APDU_CODE_EXECUTION_ERROR);
+    }
+
+    //We want to reuse extractHDPath, as it works on G_io_apdu_buffer we have to follow 
+    MEMCPY(G_io_apdu_buffer, &slot.path, sizeof(slot.path));
+    rx = sizeof(slot.path);
+    extractHDPath(rx, 0); //this puts hdPath into hdPath global variable
+
+    //We fill account and pubkey
+    MEMCPY(G_io_apdu_buffer, &slot.account, sizeof(slot.account));
+    *tx = sizeof(slot.account);
+    *tx += app_fill_pubkey(G_io_apdu_buffer + sizeof(slot.account), IO_APDU_BUFFER_SIZE - sizeof(slot.account)); 
```

Comment:
We should check the returned error here.

In `app_fill_pubkey`, it's also better not mix error codes and size of written bytes (nothing proves that the public key size 65 can't be equal to an error value). We could use an extra parameter `*int` for the size of written bytes.

Reply:
app_fill_pubkey throws if an error occurs